compose/service.py

```diff
@@ -519,6 +522,59 @@ def pull(self, insecure_registry=False):
             )
 
 
+def get_container_data_volumes(container, volumes_option):
+    """Find the container data volumes that are in `volumes_option`, and return
+    a mapping of volume bindings for those volumes.
+    """
+    volumes = []
+
+    volumes_option = volumes_option or []
+    container_volumes = container.get('Volumes') or {}
+    original_volumes = set(volumes_option + container_volumes.keys())
+
+    for volume in original_volumes:
```

Comment:
I tried to create a minimal reproduction of this failure, but when I tried with the docker cli I was not able to reproduce the failure.

https://gist.github.com/dnephin/e584a7741a514d631632

From the docker inspect they look almost identical, except you can see in the fig example the volume is not set to the path specified by `binds`.  

I tried to use mitmproxy to inspect the differences between the http requests being used, but it seems like the docker cli ignores `HTTP_PROXY` (I was able to get the requests made by fig at least).

I'm struggling to find the difference between this example and the test case. Does anyone know of a way to make the docker cli use a proxy?